### ISSUE TITLE
Potential unintentional failure for Focus Not Obscured (Enhanced), due to wording and definition

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag#2736]
The wording of [2.4.13 Focus Not Obscured (Enhanced)](https://www.w3.org/TR/WCAG22/#focus-not-obscured-enhanced) reads:
> When a [user interface component](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components) receives keyboard focus, no part of the [focus indicator](https://www.w3.org/TR/WCAG22/#dfn-focus-indicator) is hidden by author-created content.

This causes a potential problem where an item with a compliant focus indicator (per 2.4.11 Focus Appearance) could fail this AAA even though the component and the area of the focus indicator that passes Focus Appearance are not obscured at all.

This is caused because the wording of Focus Not Obscured talks about "no part of the focus indicator" instead of "no part of the sufficient focus indicator area" or words to that effect. Essentially, the second set of bullets in Focus Appearance allow for part of the focus indicator to be below contrast requirements (for things like halos, shadow, fades, etc) so long as an area of sufficient size and contrast exists, whereas 2.4.13 just points to the definition, which reads:
> pixels that are changed to visually indicate when a user interface component is in a focused state

The following two images, which are Figure 17 and 18 in the  Focus Appearance Understanding document, may help. The first uses a shadow to indicate focus around the middle of three buttons. The entire shadow is not needed to meet the contrast requirements. 
![focus-indicator-box-shadow](https://user-images.githubusercontent.com/14298245/196245618-65367180-79a1-4c66-bb08-4cd38ae58ff5.png)
The second image shows the parts of the focus indicator that do meet 3:1. 
![image](https://user-images.githubusercontent.com/14298245/196245746-e642fdeb-482c-4715-99de-e542a2bc6459.png)
If something obscured the very light pixels that 'smooth out' the shadow in the first image, it would not affect any of the pixels that are meeting the focus area. 

It follows logically that an inconsequential part of the focus indicator should not lead to a failure of Focus Not Obscured, even the enhanced version.

---
If the rephrasing suggested above does not work, another approach is to fall back to just making the enhanced version not obscure the component as opposed to its focus indicator. For instance:
> When a [user interface component](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced#dfn-user-interface-component) receives keyboard focus, no part of the component is hidden by author-created content.

 This opens up a potential situation where the a non-surrounding focus indicator could be obscured while the component itself is visible. (In the second image above, imagine if a box covered everything right up to the bottom edge of the actual button; it would make the focus indicator much harder to notice). However, this is a much smaller risk than failing good designs with the current wording.

**Comment by bruce-usab** [comment]
FWIW, I would prefer to leave as is rather than introduce a new concept of _sufficient_ focus indicator area.  One of the features of AAA SC is that they are simple to read.

2.4.13 following 2.4.12 but having "the component is not entirely hidden" be instead something like "the component is not at all hidden" seems better to me.

**Comment by detlevhfischer** [comment]
@mbgower I have to admit that even after repeated reading I don't get what the issue is. What is obscured by the shadow? Are you talking about some adjacent content (not visible in the graphic) that may be overlapped by a fading shadow?

**Comment by mbgower** [comment]
> What is obscured by the shadow? Are you talking about some adjacent content (not visible in the graphic) that may be overlapped by a fading shadow?

Ah, yes. Let me provide a low fi example
![image](https://user-images.githubusercontent.com/14298245/197843088-2a86a04a-03e2-446e-a9a2-646dece0ead2.png)

Pretend the box contains some author content. It is not obscuring the button with focus at all, and is only covering a couple of pixels of the lightest part of the drop shadow. It interferes with a user's ability to see the item with focus not at all, but it would fail the language of the AAA criteria.

**Comment by detlevhfischer** [comment]
Hm. The vontrast of the light grey just above the upper edge of the box to the box border is 9.5:1 and in no way it could be said that anything is obscured here. The only conceivable problem I see is that other content may become obsured by a focus indicator fuzzy it if placed closer to the focused element. So wouldn't be the more likely problem that (certain types of) focus indicator obscure content, rather than the other way round?

**Comment by bruce-usab** [comment]
One example which might be helpful for Understanding is a focus indicator which instead of surrounding the component (as is typical) the quite noticeable focus is on the _inside_ of the component.  Such an example might help explain why 2.4.13 is about focus indicator (and not component, as is the case with 2.4.12).

**Comment by mbgower** [comment]
> in no way it could be said that anything is obscured here. 

@detlevhfischer but my point is that it _does_ obscure some below 3:1 pixels of the focus indicator. So it fails the current wording.

**Comment by mbgower** [comment]
> the quite noticeable focus is on the inside of the component. 

@bruce-usab but if the entire item was not obscured, neither would the focus indicator be. so the change in wording would capture that.

**Comment by alastc** [comment]
> Hm. The contrast of the light grey just above the upper edge of the box to the box border is 9.5:1 and in no way it could be said that anything is obscured here. 

By the letter of the SC, it is part of the focus indicator, and it is hidden by the new content.

The question at this stage is whether it is worth updating to work around that.

It is difficult to reference another SC in the way suggested with "no part of the sufficient focus indicator area", I'm not sure how that would work.

Regarding "no part of the component is hidden", that's possible, does it make a big difference for AAA?

**Comment by alastc** [comment]
I'll to survey with something like:
<hr>

There is an over-requirement where an item with a compliant focus indicator (per 2.4.11 Focus Appearance) could fail Focus Not Obscured (Enhanced) even though the component and the area of the focus indicator that passes Focus Appearance are not obscured at all.

This is likely to affect focus indicators which have a halo or fade effect, where some of the faded area could be obscured by other content.

Is your preferred solution:

- Tp accept this, it is at AAA. 
- Update the AAA SC to only include the component in scope: "When a [user interface component](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced#dfn-user-interface-component) receives keyboard focus, no part of the component is hidden by author-created content."
- Update the SC to cross-reference focus-appearance, saying that no part of the sufficient indicator is obscured.

**Comment by alastc** [comment]
Closing after this change: https://github.com/w3c/wcag/pull/2838/files
From https://lists.w3.org/Archives/Public/w3c-wai-gl/2023JanMar/0013.html
